daemon/environment.cpp

```diff
@@ -763,7 +763,7 @@ bool verify_env(MsgChannel *client, const string &basedir, const string &target,
     reset_debug();
     chdir_to_environment(client, dirname, user_uid, user_gid);
     execl("bin/true", "bin/true", (void*)NULL);
-    log_perror("execl failed");
+    log_perror("bin/true");
```

Comment:
This will lead to error message "bin/true(Error: No such file or directory)", which is confusing. If you think the binary to execute is important there, please add it to the existing error message.

Reply:
Done.

Reply:
I commented on this last one, but I meant it as a generic comment that affects all the changes. It should be something like log_perror("execl failed (" + argv[ 0 ] + ")"), with the necessary std::string handling.

Reply:
Okay. I'm using `std::ostringstream` for concatenating the strings.